druid-shell/src/platform/x11/application.rs

```diff
@@ -525,36 +549,57 @@ fn poll_with_timeout(conn: &Rc<XCBConnection>, idle: RawFd, timeout: Instant) ->
 
     // We start with no timeout in the poll call. If we get something from the idle handler, we'll
     // start setting one.
-    let mut poll_timeout = -1;
+    let mut honor_idle_timeout = false;
     loop {
         fn readable(p: PollFd) -> bool {
             p.revents()
                 .unwrap_or_else(PollFlags::empty)
                 .contains(PollFlags::POLLIN)
         };
 
+        // Compute the deadline for when poll() has to wakeup
+        let deadline = if honor_idle_timeout {
+            Some(earliest_timeout)
+        } else {
+            timer_timeout
+        };
+        // ...and convert the deadline into an argument for poll()
+        let poll_timeout = if let Some(deadline) = deadline {
+            if deadline < now {
+                0
```

Comment:
Why not just break here?

Reply:
A combination of "historic reasons" and "I do not follow understand the idle drawing code". The previous code definitely did not return early and might have went into a short busy loop. I think. I am not sure.

I'll just change it to a break (and also to `<=` instead of `<`, but that shouldn't make much of a difference).